### ISSUE TITLE
`sendline()` should have semantics of `sendall()` instead of `send()`.

### ISSUE THREAD

**Issue by MrMino** [pexpect/pexpect#515]
_TL;DR: `sendline()` that returns a number of sent bytes doesn't make sense, when line separator is unknown to the code that uses that method._

The expected semantics of `send()` methods (e.g. `sock.send()`) dictate, that the code using it should handle situations, where the returned value (number of bytes sent) is less than the length of the data to send.

This would be used to infer how much data requires retransmission when an error occurs.

Given that `sendline()` is roughly equivalent to `return send(s + lineseparator)`, in order for the return value of it to be of any use to the code that uses it, that code needs to know the length of `s + lineseparator`.

That means, that in order to use `sendline()`, I need to know `os.linesep` myself anyway, making the whole thing redundant. After all, the whole point of that method (from what I understand from its documentation) is for it to be a convenience that covers the knowledge of the line separator for the platform for me.

I propose the following:

1. adding `sendall()` method to `spawn`, with semantics that follow `sock.sendall()`
2. making `sendline()` use `sendall()` instead of `send()`. 

I'd be happy to help, if you think it is a good idea :).

**Comment by MrMino** [comment]
BTW. I've just checked, the impact of removing `return` from both popen's and pty's `spawn.sendline()` is none. All tests (apart from FSM test, which for some reason fails on me either way) pass whether or not `sendline()` actually returns anything.
